### DISCUSSION TITLE
Adding OMPlot to openmodelica cask

### DISCUSSION THREAD

**Discussion by l0r3m1psum** [Homebrew/discussions#2040] [Casks]
Hello, this is my first post here and I'm not sure if I'm using this right.

Today I was looking at the formula for [openmodelica](https://github.com/Homebrew/homebrew-core/blob/master/Formula/openmodelica.rb) because I needed it to install the `OMPlot` command. On line [70](https://github.com/Homebrew/homebrew-core/blob/master/Formula/openmodelica.rb#L70) I've noticed that `omplot` is not passed as an argument to `make` because of an issue cited at line [66](https://github.com/Homebrew/homebrew-core/blob/master/Formula/openmodelica.rb#L66), and same for `omparser`. So I looked at both issues ([7240](https://github.com/OpenModelica/OpenModelica/pull/7240) and [7247](https://github.com/OpenModelica/OpenModelica/pull/7247)) and both seems to have been solved. I've tried to change the formula locally adding both arguments to make. Trying to run `brew reinstall --build-from-source --debug --verbose openmodelica` gives, in short, the following error: `clang: error: no such file or directory: 'x86_64'`. After some digging I've discovered that clang is not looking for a directory but is missing the `-arch` flag before the `x86_64` string. This seems to be because of a makefile generated by a command, namely

```
# The CWD of the process is: /private/tmp/openmodelica-20210825-14034-1ozvvwq/OMPlot/OMPlot/OMPlotGUI/
$ sh /private/tmp/openmodelica-20210825-14034-1ozvvwq/OMPlot/qmake.sh "CONFIG+=release" OMPlotLib.pro -o Makefile.lib
$ /Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile.lib
```

In fact adding `-arch` in `Makefile.lib` wherever needed fixed the problem.

So this looks like a openmodelica's problem. But I don't think they'll keep a bug in their build system. So does anybody have any idea of how to fix this problem? It would be really nice to be able to install this two commands with the openmodelica formula, given that the issue because they were left out are both solved.

**Comment by SMillerDev** [comment]
> So this looks like a openmodelica's problem. But I don't think they'll keep a bug in their build system.

Lots of projects don't really test on macOS because it's hard to get CI for it. I'd just contact them and see what they think.

**Reply by l0r3m1psum** [reply]
Hmm makes sense, I've tried to build OMPlot from source and looks like it doesn't work on macOS but omparser seem to build fine so I think it can be put in the formula.

**Reply by cho-m** [reply]
Neither of the PR fixes appear to be available in a stable release:
- https://github.com/OpenModelica/OpenModelica/commit/2e75be3844d32a997e3811f00d3ec105726c9f5b
- https://github.com/OpenModelica/OpenModelica/commit/359b464345e08cfae25e1bd352d65a3927f39c5a

Unless they backported them, the commits are only in beta/dev releases shown in tags (v1.19.0-dev, v1.18.0-dev.beta1).

**Reply by l0r3m1psum** [reply]
You are right. So we have wait until a new stable release appear.  Thank you to both.
